spicy/toolchain/src/compiler/codegen/parsers/types.cc

```diff
@@ -167,7 +167,12 @@ struct Visitor : public hilti::visitor::PreOrder<Expression, Visitor> {
         return performUnpack(destination(t), t, t.width() / 8, {state().cur, fieldByteOrder()}, t.meta(), is_try);
     }
 
-    result_t operator()(const hilti::type::Void& /* t */) { return hilti::expression::Void(); }
+    result_t operator()(const hilti::type::Void& /* t */) {
+        // Even though we we do not store parsed data, we still need to consume
+        // data in the input stream so that e.g., `&size` checks can work.
+        pb->advanceInput(builder::size(state().cur));
```

Comment:
This comment feels a bit misleading because I think it's really *only* with `&size` where skipping ahead over the all current input is the right thing to do. Are we already rejecting `void` fields that don't have any attribute? If we don't then this would act like `&eod` for them, right? If we want that, we should make it explicit and require `: void &eod`. 

In other words, let's make sure we validate that only supported cases can arrive here, and update the comment to make that assumption clear.

Reply:
Thanks for the careful reading, this indeed introduced a bug.

The current behavior is for `void` fields w/o explicit `&size` is to consume nothing which is required behavior if used in unit switch statements. I added a new test checking that unit-level `void` fields do indeed consume nothing, and also introduced a validator to reject any attributes other than `&size` for `void` fields (e.g., `&max-size` would introduce off-by-one errors on `void` fields).

While I was at it I also added code to reject `void` fields with names.